core/types/tx_setcode.go

```diff
@@ -70,12 +70,13 @@ type SetCodeTx struct {
 
 // SetCodeAuthorization is an authorization from an account to deploy code at its address.
 type SetCodeAuthorization struct {
-	ChainID uint256.Int    `json:"chainId" gencodec:"required"`
-	Address common.Address `json:"address" gencodec:"required"`
-	Nonce   uint64         `json:"nonce" gencodec:"required"`
-	V       uint8          `json:"yParity" gencodec:"required"`
-	R       uint256.Int    `json:"r" gencodec:"required"`
-	S       uint256.Int    `json:"s" gencodec:"required"`
+	ChainID   uint256.Int    `json:"chainId" gencodec:"required"`
+	Address   common.Address `json:"address" gencodec:"required"`
+	Nonce     uint64         `json:"nonce" gencodec:"required"`
+	V         uint8          `json:"yParity" gencodec:"required"`
+	R         uint256.Int    `json:"r" gencodec:"required"`
+	S         uint256.Int    `json:"s" gencodec:"required"`
+	authority common.Address `json:"-"`
```

Comment:
The `authority` field is not thread-safe.

The similar cached fields in transaction use atomic,

```go
	// caches
	hash atomic.Pointer[common.Hash]
	size atomic.Uint64
	from atomic.Pointer[sigCache]
```

Although I am not entirely confident about this change, it affects only a single call site in the main block processing that uses the `Authority` function. 

cc @lightclient wdyt?

Reply:
The CI failed after using atomic.Pointer: https://ci.appveyor.com/project/ethereum/go-ethereum/builds/51528023/job/73bh6kacgct60dil . Should we change `SetCodeTx.AuthList` to `[]SetCodeAuthorization` instead?